### DISCUSSION TITLE
How to check the results

### DISCUSSION THREAD

**Discussion by fu-zzy** [BindsNET/bindsnet#531] [Q&A]
I am a first time student of artificial intelligence. I'm trying to learn using this SNN. My question is: I have run eth_mnist, how do I check the resulting data? I ran it in windows10 standard command prompt. I was able to see the result plots during the process, but after complete all of them were hidden.

Finally, thank you for publishing such a great project!
I will continue to use it.　From Japan
![スクリーンショット (8)_LI](https://user-images.githubusercontent.com/57170726/147304456-1c46a653-e361-487c-b90a-aced63895283.jpg)
.

**Comment by Hananel-Hazan** [comment]
Thank you for using BindsNET. 
According to the screenshot you shared, the accuracy of the model is 79% using the proportion method and 78% using the all activity method. The `eth_mnist` script produce image on the fly, in order to save the images, you need to add `save=` to the plot command similar to the `SOM_LM-SNNs.py` .

**Reply by fu-zzy** [reply]
Hi.Thanks for the answer. I followed the instructions and was able to save weights,performance and asymmetry.I tried to save weights, performance, and asymmetry, but I couldn't save spike, input, and voltage. I want to generate an image like the below, how can I save it?

Also, when I run the example in the name of example/mnist, it runs on cuda, but when I run the code I rewrote a little, it runs on cpu, even though I didn't rewrite the GPU settings.

I'm sorry for repeating something so simple. Thanks for your help.
![スクリーンショット (13)](https://user-images.githubusercontent.com/57170726/148689499-54d358fb-017a-4ce4-a0a1-76788f424e05.png)

**Reply by Hananel-Hazan** [reply]
Hi.

To save the spikes and the voltage values, you need to save the content of the network monitors. In the `eth_mnist.py` the variable `spike_record` contain the spikes of the current iterations, and `exc_voltages` and `inh_voltages` are the voltage values for the excitatory and the inhibitory neurons. 
To save the figure, add `save= FILENAME_AND_PATH` to the plotting function and it will save the figure to the FILENAME_AND_PATH that you specified. For example the plotting functions in `eth_mnist.py` are in lines 267,268,268.

About the reason why the scrip use the CPU and not the GPU, that is hard to tell, it depends on the changes made to the script or to your running environment.

Hope its helps

**Reply by fu-zzy** [reply]
Thanks for the answer.

I tried adding the code as you did, but I can't get it to run. I wonder why.
I didn't change anything except the save command in the image below, and the path is correct.
By the way, I was able to run it after commenting out input, spike, and voltage.

Sorry for asking again and again on such rudimentary details. I'm waiting for your answer.
![スクリーンショット (14)](https://user-images.githubusercontent.com/57170726/148882309-a862996d-60f0-4f32-bd74-eeec74959de3.png)
.

**Reply by Hananel-Hazan** [reply]
According to the screenshot, only `plot_voltages` function failed to run, that mean that all other function save the image correctly?
The `plot_voltages` don't have the `save`, option, if you need it, try to implement it. If you cant implement it yourself, open an issue and one of the students may be in the future add this functionality.
